enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,214 @@
+by the customer.
+
+## Motivation
+
+Combined with `FailureDomains`, it will offer more resiliency by allowing to configure compute and control
+plane nodes across multiple subnets for on-premise IPI deployments.
+It also offers more flexibility by giving more control to the customer so they can use their own loadbalancer
+solution, instead of the built-in that the IPI workflow installs and manage how the VIPs are routed (e.g. via BGP).
+It also help to overcome some performance issues with the self-hosted loadbalancer. when it comes to scalability.
+
+### User Stories
+
+As a deployer of a IPI OpenShift cluster, I want to place my control plane nodes on 3 or
+more subnets at installation time. Therefore I need to be able to configure an external load balancer for
+the VIPs used on the cluster control plane.
+
+### Goals
+
+Allow to replace the internal loadbalancer by an external one and support this scenario when the customer
+prefers to maintain it themselves. Later we provide more details on what is expected from the customer to do.
+
+### Non-Goals
+
+* Making this scenario by default at some point. The internal loadbalancer for on-premise platforms will
+  remain the HAproxy/Keepalived/CoreDNS stack.
+* Supporting a mix of both (internal + external loadbalancers for the API & Ingress VIPs), customers
+  will have to choose a scenario.
+
+## Proposal
+
+We will make some change in `install-config.yaml` so customer can configure the
+deployment to use an external loadbalancer if they want.
+Then the components which interact with networking will have to check this
+new field and handle this scenario appropriately.
+
+### Workflow Description
+
+1. The customer will configure their external loadbalancer(s), which includes configuring the VIPs
+   for API and Ingress and also configure the loadbalancer tool that is being used (e.g. HAproxy, F5, etc)
+2. The customer will write the `install-config.yaml` file and set the new `loadBalancerType` platform specific field to `External`, and
+   will set the fields for `apiVIP` and `ingressVIP`.
+3. In the installer, if the platform does support external loadbalancer, some validation will be run
+   to check that the VIPs can already be resolved (the validation already exists for vsphere, we need to
+    make it re-usable). The new field for `LoadBalancerType` will be set in `PlatformSpec`.
+4. In machine-config-operator, if LB type is external, the internal loadbalancer will not be
+   deployed.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="Internal";"External"
+type PlatformLoadBalancerType string
+const (
+  // LoadBalancerTypeExternal is a loadbalancer that is managed outside of the cluster by the customer.
```

Comment:
do we want to keep `LoadBalancerTypeExternal` and `LoadBalancerTypeInternal` as they are?